### PR TITLE
Extend tab hit box to top of tabstrip when maximized

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/34038

Currently, tab views are inset vertically from the tab strip in order to create the upper and lower margins. With this change, tab views occupy the full vertical space of the tab strip; the upper and lower margins are created in the tab path-drawing procedure. This allows us to extend the hit boxes to the top of the screen when the window is maximized.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/views/tabs/brave_tab_style_views.inc.cc [hunk 1]
```diff
@@ -113,6 +104,26 @@ SkPath BraveVerticalTabStyle::GetPath(
...
+    aligned_bounds.Inset(
+        gfx::InsetsF::VH(brave_tabs::kHorizontalTabVerticalSpacing * scale,
+                         brave_tabs::kHorizontalTabInset * scale));
+
+    // For maximized and fullscreen windows, extend the tab hit test to the top
+    // of the tab, encompassing the top spacing. This makes it easy to click on
+    // tabs by moving the mouse to the top of the screen.
+    if (path_type == TabStyle::PathType::kHitTest &&
+        (tab()->GetWidget()->IsMaximized() ||
+         tab()->GetWidget()->IsFullscreen())) {
```
Comment (sangwoo108) [inline browser/ui/views/tabs/brave_tab_style_views.inc.cc:121]: Can we replace this with `ShouldExtendHitTest()`?
```suggestion
        ShouldExtendHitTest()) {
```

File: browser/ui/views/tabs/brave_tab_style_views.inc.cc [hunk 2]
```diff
@@ -113,6 +104,26 @@ SkPath BraveVerticalTabStyle::GetPath(
...
+      aligned_bounds.Inset(gfx::InsetsF::TLBR(
```
Comment (sangwoo108) [inline browser/ui/views/tabs/brave_tab_style_views.inc.cc:122]: optional: There's gfx::Rect::Outset() and would be perfect for this, but I'm okay with the current version 👍

**Review by sangwoo108 — APPROVED** [review]
Nice work! Looks so much better 👍

File: browser/ui/views/tabs/brave_tab_style_views.inc.cc (hunk 1 above)
Comment (zenparsing) [reply browser/ui/views/tabs/brave_tab_style_views.inc.cc]: Done

File: browser/ui/views/tabs/brave_tab_style_views.inc.cc (hunk 2 above)
Comment (zenparsing) [reply browser/ui/views/tabs/brave_tab_style_views.inc.cc]: Done

File: browser/ui/brave_layout_constants.cc [hunk 3]
```diff
@@ -21,9 +21,9 @@ absl::optional<int> GetBraveLayoutConstant(LayoutConstant constant) {
   //     ui::MaterialDesignController::IsNewerMaterialUi();
   switch (constant) {
     case TAB_HEIGHT: {
-      const int tab_height = HorizontalTabsUpdateEnabled() ? 36 : 30;
-      return (touch ? 41 : tab_height) +
-             GetLayoutConstant(TABSTRIP_TOOLBAR_OVERLAP);
+      const int tab_height =
+          HorizontalTabsUpdateEnabled() ? 44 : (touch ? 41 : 30);
```
Comment (cdesouza-chromium) [inline browser/ui/brave_layout_constants.cc:25]: Do you think it would be appropriate to give these values descriptive constant names?

**Review by mkarolin — APPROVED** [review]
`chromium_src`++

**Review by cdesouza-chromium — APPROVED** [review]
LGTM

File: browser/ui/brave_layout_constants.cc (hunk 3 above)
Comment (zenparsing) [reply browser/ui/brave_layout_constants.cc]: The 44 value, I think so. The other values (existing code used when the flag is off) I'll leave as is.

**Pull request merged**